src/lua/luaLoggerBinding.c

```diff
@@ -740,71 +666,244 @@ int Lua_ReadOBD2(lua_State *L)
-
-    ChannelConfig cc;
-    strlcpy(cc.label, lua_tostring(L, 1), DEFAULT_LABEL_LENGTH);
-    cc.sampleRate = encodeSampleRate((unsigned short) lua_tointeger(L, 2));
-    cc.precision = args >= 3 ? lua_tointeger(L, 3) :
-                   DEFAULT_CHANNEL_LOGGING_PRECISION;
-    cc.min = args >= 4 ? lua_tointeger(L, 4) : DEFAULT_CHANNEL_MIN;
-    cc.max = args >= 5 ? lua_tointeger(L, 5) : DEFAULT_CHANNEL_MAX;
-    strlcpy(cc.units, args >=6 ? lua_tostring(L, 6) : DEFAULT_CHANNEL_UNITS,
-            DEFAULT_UNITS_LENGTH);
-    lua_pushinteger(L, create_virtual_channel(cc));
+        const size_t args = lua_gettop(L);
+        if (args < 2 || args > 6)
+                return incorrect_arguments(L);
 
-    return 1;
+        ChannelConfig cc;
+        channel_config_defaults(&cc);
+
+        switch (validate_channel_config_label(lua_tostring(L, 1))) {
+        case CHAN_CFG_STATUS_OK:
+                strcpy(cc.label, lua_tostring(L, 1));
+                break;
+        case CHAN_CFG_STATUS_NO_LABEL:
+                luaL_error(L, "Label is empty");
+        case CHAN_CFG_STATUS_LONG_LABEL:
+                luaL_error(L, "Label is too long");
+        default:
+                goto bug;
+        }
+
+        cc.sampleRate = encodeSampleRate((unsigned short) lua_tointeger(L, 2));
+        if (SAMPLE_DISABLED == cc.sampleRate)
+                return luaL_error(L, "Unsupported sample rate");
+
+        switch(args) {
+        case 6:
+                switch (validate_channel_config_units(lua_tostring(L, 6))) {
+                case CHAN_CFG_STATUS_OK:
+                        strcpy(cc.units, lua_tostring(L, 6));
+                        break;
+                case CHAN_CFG_STATUS_NO_UNITS:
+                        return luaL_error(L, "Units is empty");
+                case CHAN_CFG_STATUS_LONG_UNITS:
+                        return luaL_error(L, "Units is too long");
+                default:
+                        goto bug;
+                }
+        case 5:
+                cc.max = lua_tointeger(L, 5);
+        case 4:
+                cc.min = lua_tointeger(L, 4);
+        case 3:
+                cc.precision = lua_tointeger(L, 3);
+        }
+
+        switch(validate_channel_config(&cc)) {
+        case CHAN_CFG_STATUS_OK:
+                break;
+        case CHAN_CFG_STATUS_MAX_LT_MIN:
+                return luaL_error(L, "Max is less than min");
+        default:
+                goto bug;
+        }
+
+        const int chan_id = create_virtual_channel(cc);
+        if (INVALID_VIRTUAL_CHANNEL == chan_id)
+                return luaL_error(L, "Unable to create channel. "
+                                     "Maximum channels reached.");
+
+        lua_pushinteger(L, chan_id);
+        return 1;
+
+bug:
+        return luaL_error(L, "BUG. Should never get here.  "
+                             "Please inform AutosportLabs");
 }
 
-int Lua_SetVirtualChannelValue(lua_State *L)
+static int lua_get_virtual_channel(lua_State *ls)
```

Comment:
So would you prefer I change it to the no longer acceptable naming standard (Camel Case) or should I alter all the method names?  I didn't do the latter because I felt it would cause unnecessary churn/noise.  But I would be happy to do it and get it out of the way.